util/types/time.go

```diff
@@ -949,7 +949,7 @@ func (d Duration) MicroSecond() int {
 // ParseDuration parses the time form a formatted string with a fractional seconds part,
 // returns the duration type Time value.
 // See http://dev.mysql.com/doc/refman/5.7/en/fractional-seconds.html
-func ParseDuration(str string, fsp int) (Duration, error) {
+func ParseDuration(str string, fsp int) (Duration, bool, error) {
```

Comment:
the second return value is `true` only when the returned error is `ErrTruncatedWrongVal`, can we keep the return values to `(Duration, error)`, and check `ErrTruncatedWrongVal` outside ?

Reply:
@zz-jason If I check errors inside func ParseDuration, then I needs StatementContext to deal with SQLMODE.
Any suggestions?

Reply:
No, I mean maybe we don't need to check the error inside function `ParseDuration()`, you can just return the error and let the function caller to handle it(ignore or throw it).